### PR TITLE
Mithminite scythe rework

### PR DESCRIPTION
**Pull request by TabakaSIM**
Mithminite scythe reworked, that includes:
- all 51 aspects have effect!
- buffed projectiles
- augmentation mechanic
- questbook entry

> What was tested:
> - all effects
> - most "performance heavy" combination: aer, fluctus, vacuos, draco, rattus, mana, sonus, exitium

> What wasn't tested:
> - on server - i can't join on my local server :(

> What is missing:
> - translations .ru and .zh
> - gif with cool looking projectile effect (i'll add it later)

### CODE REVIEW

**Comment by Krutoy242** [comment]
Thank you for the contribution! I will take closer look later.

**Comment by Krutoy242** [comment]
Did you tested if scythe will keep it's enchants or other nbt tags like Random Tweaker anchor on upgrading?

**Comment by Krutoy242** [comment]
Maybe I got this wrong, but, is scythe storing it's upgrades worth lore nbt? As I recall, Random Tweaker have special lore tags that could change lore of items. Did you test if it would exploit or destroy your nbt?

**Comment by Krutoy242** [comment]
Please move crafting hints to scythe zs file. I did a mistake by adding modded hints to hints.zs file itself and want to keep it as library that other files using. I will need to refactor that later but for now you could just nice each hint to respective mechanic function definition.

**Comment by Krutoy242** [comment]
Ok, since I don't know other way to unhide this comments I will show them as images 😅








41-397572927bc3)

**Comment by TabakaSIM** [comment]
> Did you tested if scythe will keep it's enchants or other nbt tags like Random Tweaker anchor on upgrading?

Fixed (Guess I fallen in love with deepUpdate from Zenscripts)


> Maybe I got this wrong, but, is scythe storing it's upgrades worth lore nbt? As I recall, Random Tweaker have special lore tags that could change lore of items. Did you test if it would exploit or destroy your nbt?

Honestly, I tried to use them, but haven't worked at all
 

 > It's squared distance, 7 blocks. Isn't it too small?

 It's actually normal distance (checked crafttweaker code). Sq stands for euclidean distance i guess

File: scripts/mods/thaumadditions.zs [hunk 1]
```diff
@@ -1662,14 +1646,11 @@ recipes.addShapeless("REMOVEaugmentMithminiteScythe",<thaumadditions:seal_symbol
...
-    val index = lore.length;
-    var lorem as string[] = [];
-    var result = <thaumadditions:mithminite_scythe>.withTag(item.tag - {display : "just delete"});
-    if(lore.length > 1){for i in 0 to (lore.length - 1) {lorem += lore[i];}}
+    var lore = item.tag.display.Lore;
 
-    return isNull(item.tag.display.Name) ? result.withLore(lorem) : result.withTag({display : {Name: item.tag.display.Name}}).withLore(lorem); 
+    lore = lore.deepUpdate([lore[lore.length - 1]],REMOVE);
 
+    return item.updateTag({display:{Lore : lore}});
```
Comment (Krutoy242) [inline scripts/mods/thaumadditions.zs:1653]: Are you sure this will not remove custom name from Scythe?

File: scripts/do/mithminite_scythe.zs [hunk 2]
```diff
@@ -278,7 +215,7 @@ function dracoBreath(target as IEntityLivingBase) as void{
 
     for entity in listAllEntities {
         if(isNull(entity)
-        || isNull(entity.definition)
+        || entity instanceof IPlayer
```
Comment (Krutoy242) [inline scripts/do/mithminite_scythe.zs:218]: Now this line is duplicated

**Review by Krutoy242 — CHANGES_REQUESTED** [review]

**Comment by Krutoy242** [comment]
I pressed "Request Changes". Could you see comments now?

**Comment by TabakaSIM** [comment]
Yes, now i can see them all

**Pull request merged**